src/sc_client/client/_payload_factory.py

```diff
@@ -137,42 +140,55 @@ def __call__(self, *params: ScIdtfResolveParams):
+        elif isinstance(template, str):
+            payload_template = template
         else:
-            for triple in template.triple_list:
-                items = [
-                    triple.get(common.SOURCE),
-                    triple.get(common.EDGE),
-                    triple.get(common.TARGET),
-                ]
-                payload_triple = [process_triple_item(item) for item in items]
-                is_required = triple.get(common.IS_REQUIRED)
-                if not is_required:
-                    payload_triple.append({common.IS_REQUIRED: is_required})
-                payload.append(payload_triple)
-        return payload
+            payload_template = self._process_template(template)
+
+        payload_params = {}
+        if params is not None:
+            for alias, addr in params.items():
+                if isinstance(addr, ScAddr):
+                    payload_template.append({alias: addr.value})
+                else:
+                    payload_template.append({alias: str(addr)})
+        return {common.TEMPLATE: payload_template, common.PARAMS: payload_params}
+
+    def _process_template(self, template: ScTemplate):
+        raise NotImplementedError
 
 
-class TemplateGeneratePayloadCreator(BasePayloadCreator):
-    def __call__(self, template: ScTemplate | str, params: ScTemplateGenParams, *_):
+class TemplateSearchPayloadCreator(TemplatePayloadCreator):
+    def _process_template(self, template: ScTemplate):
         payload_template = []
-        if isinstance(template, str):
-            payload_template = template
-        else:
-            for triple in template.triple_list:
-                items = [
-                    triple.get(common.SOURCE),
-                    triple.get(common.EDGE),
-                    triple.get(common.TARGET),
-                ]
-                payload_template.append([process_triple_item(item) for item in items])
-        payload_params = {alias: addr.value for alias, addr in params.items()}
-        payload = {common.TEMPLATE: payload_template, common.PARAMS: payload_params}
-        return payload
+        for triple in template.triple_list:
+            items = [
+                triple.get(common.SOURCE),
+                triple.get(common.EDGE),
+                triple.get(common.TARGET),
+            ]
+            payload_triple = [process_triple_item(item) for item in items]
+            payload_template.append(payload_triple)
+        return payload_template
+
+
+class TemplateGeneratePayloadCreator(TemplatePayloadCreator):
+    def _process_template(self, template: ScTemplate):
+        payload_template = []
+        for triple in template.triple_list:
+            items = [
+                triple.get(common.SOURCE),
+                triple.get(common.EDGE),
+                triple.get(common.TARGET),
+            ]
+            payload_template.append([process_triple_item(item) for item in items])
+        return payload_template
```

Comment:
Is there any differences in TemplateGeneratePayloadCreator and TemplateSearchPayloadCreator? As I see, logic is the same

Reply:
There are differences